app/controllers/planning_applications/neighbour_letters_controller.rb

```diff
@@ -82,12 +82,14 @@ def neighbour_params
     end
 
     def update_letter_statuses
-      return if @consultation.neighbour_letters.none?
+      letters = @consultation.neighbour_letters.includes(:neighbour).where.not(status: "received")
+
+      notify_key = @planning_application.local_authority.notify_api_key || Rails.configuration.default_notify_api_key
 
       # This is not ideal for now as will block the page loading, if it becomes a problem this would be
       # a good place for optimisation
-      @consultation.neighbour_letters.each do |letter|
-        letter.update_status unless letter.status == "received"
+      letters.each do |letter|
+        letter.update_status(notify_key)
```

Comment:
We need to move this to a background job, it's very slow when there are many letters. (Out of scope for this PR but i'll add a chore in trello)

Reply:
Preloading neighbour for now to remove n+1 and setting the notify key once here instead of calling the `notify_api_key` method (which makes a query) for every iteration when updating the status